eigensync/src/lib.rs

```diff
@@ -0,0 +1,243 @@
+            request_id,
+            error,
+        })) => {
+            let sender = match response_map.remove(&request_id) {
+                Some(sender) => sender,
+                None => {
+                    println!("No sender for request id {:?}", request_id);
+                    return Ok(());
+                }
+            };
+
+            let _ = sender.send(Err(error.to_string()));
+        }
+        SwarmEvent::ConnectionEstablished { peer_id: _peer_id, .. } => {
+            // send the connection established signal
+            if let Some(sender) = connection_established.take() {
+                let _ = sender.send(());
+            }
+        },
+        other => eprintln!("Received event: {:?}", other),
+    })
+}
+
+pub struct EigensyncHandle<T: Reconcile + Hydrate + Default + Debug> {
+    pub document: AutoCommit,
+    sender: UnboundedSender<ChannelRequest>,
+    _marker: PhantomData<T>,
+}
+
+impl<T: Reconcile + Hydrate + Default + Debug> EigensyncHandle<T> {
+    pub async fn new(server_addr: Multiaddr, server_id: PeerId, keypair: identity::Keypair) -> anyhow::Result<Self> {
+
+        let mut swarm = SwarmBuilder::with_existing_identity(keypair)
+            .with_tokio()
+            .with_tcp(
+                tcp::Config::default(),
+                noise::Config::new,
+                yamux::Config::default,
+            )
+            .context("Failed to create TCP transport")?
+            .with_behaviour(|_| Ok(client()))
+            .context("Failed to create behaviour")?
+            .with_swarm_config(|cfg| cfg.with_idle_connection_timeout(Duration::MAX))
+            .build();
+
+        swarm.add_peer_address(server_id.clone(), server_addr.clone());
+        println!("Dialing server at {}/{}", server_addr, server_id);
+
+        swarm.dial(server_id).context("Failed to dial")?;
+        
+        let (sender, receiver) = unbounded_channel();
+        let (connection_ready_tx, connection_ready_rx) = oneshot::channel();
+
+        task::spawn(async move {
+            SyncLoop::new(receiver, swarm, connection_ready_tx).await.run(server_id).await.unwrap();
+        });
+
+        connection_ready_rx.await.context("Failed to establish connection")?;
+
+        let mut document = AutoCommit::new().with_actor(ActorId::random());
+        let state = T::default();
+        reconcile(&mut document, &state).unwrap();
+
+        Ok(Self { document, _marker: PhantomData , sender})
+    }
+
+    pub fn get_changes(&mut self) -> Vec<Change> {
+        self.document
+            .get_changes(&[])
+            .iter()
+            .map(|c| (*c).clone())
+            .collect()
+    }
+
+    pub fn get_document_state(&mut self) -> anyhow::Result<T> {
+        hydrate(&self.document).context("Failed to hydrate document")
+    }
+
+    pub async fn modify(&mut self, f: impl FnOnce(&mut T) -> anyhow::Result<()>) -> anyhow::Result<()> {
+        self.save_updates_local(f).context("Failed to save updates local")?;
```

Comment:
It seems that this function is simply an alias for `save_updates_local`. Remove this one and rename `save_updates_local` to `modify´